File: frontend/src/services/accountService.ts
import { apiClient } from "./api";
import {} from "../models/LoginResponse";
import type { Account } from "../models/Account";
import type { ApiResponse } from "../models/ApiResponse";

export class AccountService {
  private readonly baseUrl = "/accounts";

  async getAccountById(id: number): Promise<ApiResponse<Account>> {
    return apiClient.get(`${this.baseUrl}/${id}`);
  }

  async getAccountByEmail(email: string): Promise<ApiResponse<Account>> {
    return apiClient.get(`${this.baseUrl}/email/${email}`);
  }

  async updateAccount(
    id: number,
    account: Partial<Account>
  ): Promise<ApiResponse<Account>> {
    return apiClient.put(`${this.baseUrl}/${id}`, account);
  }

  async verifyCurrentPassword(
    accountId: number,
    currentPassword: string
  ): Promise<ApiResponse<boolean>> {
    return apiClient.post(`${this.baseUrl}/${accountId}/verify-password`, {
      currentPassword,
    });
  }

  async resetPassword(data: {
    accountId: number;
    currentPassword: string;
    newPassword: string;
  }): Promise<ApiResponse<void>> {
    return apiClient.post(`${this.baseUrl}/${data.accountId}/reset-password`, {
      currentPassword: data.currentPassword,
      newPassword: data.newPassword,
    });
  }
}

export const accountService = new AccountService();

File: backend/src/main/java/com/flightmanagement/service/AccountService.java
package com.flightmanagement.service;

import com.flightmanagement.dto.AccountDto;
import com.flightmanagement.dto.RegisterDto;
import com.flightmanagement.entity.Account;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

// AccountService.java (Interface)
public interface AccountService {
    AccountDto createAccount(RegisterDto dto);

    AccountDto getAccountById(Integer id);

    AccountDto getAccountByEmail(String email);

    Account getAccountByName(String accountName);

    List<AccountDto> getAllAccounts();

    Page<AccountDto> getAllAccountsPaged(Pageable pageable);

    AccountDto updateAccount(Integer id, AccountDto dto);

    void deleteAccount(Integer id);

    boolean existsByEmail(String email);

    List<AccountDto> getAccountsByType(Integer accountType);

    boolean verifyCurrentPassword(Integer accountId, String currentPassword);
    void resetPassword(Integer accountId, String currentPassword, String newPassword);
}


